### ISSUE TITLE
WYSIWYG Table Generator & Editor for Markdown Mode

### ISSUE THREAD

**Issue by tuaris** [BookStackApp/BookStack#315]
### For Feature Requests

Desired Feature: WYSIWYG Table Generator & Editor for Markdown Mode

##### Expected Behavior
When in markdown editor mode.   Have the ability to create and edit tables in the same way when using the HTML editor. 

##### Actual Behavior
Current behavior requires you to edit/create tables manually which can get complex and confusing.

**Comment by tmikaeld** [comment]
The most impressive MD editor there is right now happens to be open source:
https://github.com/benweet/stackedit

Adding that to Bookstack would make it absolutely amazing to write in.

**Comment by Kovah** [comment]
I would like to second that. Tables take very long to create but what bothers me most are simple editor functions like making text bold. Support for keyboard shortcuts like `cmd` + `b` for making the selected text bold would be super awesome. I think using an editor like the one mentioned by @tmikaeld offers this functionality by default.

**Comment by ssddanbrown** [comment]
@Kovah Feel free to open an issue for adding a bold shortcut to the markdown editor.

If not yet seen, Available shortcuts can [be seen here](https://www.bookstackapp.com/docs/user/markdown-editor/).

**Comment by derek-shnosh** [comment]
I realize this is a bit of a necro, but felt it worth throwing my support for.

I do like how the MD editor automatically makes the first row `<thead>`, Is there a way to do this without editing source code in the WYSIWYG editor?
Either way, that stackedit MD editor looks impressive.

**Comment by Vicvinegar0** [comment]
After trying to deal with large tables (500+ rows), TinyMCE's performance is very poor. ToastUI seems to handle that much better. So I also support the idea of a new/addition editor if that's a possibility in the future.

**Comment by tmikaeld** [comment]
Tables will never really be good in a wysiwyg editor, I'd suggest having a "Table" mode that's separate and use [Handsontable](https://github.com/handsontable/handsontable) (MIT Licensed) for that. 

It can easily handle thousands of rows and csv import/export.

**Comment by Vicvinegar0** [comment]
@tmikaeld I agree, that's ideal. The tables in these basic editors can't handle a lot and a full on spreadsheet function would be great. Will have to leave my sheets in Excel until that.

**Comment by derek-shnosh** [comment]
To clarify the reason for my preference; my support for the WYSIWYG editor (e.g. ToastUI) is because it saves/writes the table in MD syntax, so the source document file isn't bloated.

However the functionality of Handsontable is intriguing; and I'd throw my support in it's corner if I were able to create a table separately and reference/embed it in a document. So the document file doesn't have the source syntax/code of the sheet cluttering up the content.

**Comment by Wookbert** [comment]
**Alternative idea:** Why not display Google Sheets inline? The excerpt in the viewing area should be scrollable (horizontal/vertical) and perhaps zoomable. No inline-editing. Instead a pencil button opens the sheet on Google’s website.

**Comment by tmikaeld** [comment]
@Wookbert That's convenient of course, but for those that don't want to give their data to Google, that's not an alternative.

**Comment by Wookbert** [comment]
> @Wookbert That's convenient of course, but for those that don't want to give their data to Google, that's not an alternative. But it would be nice as an option.

Well, I'm sure there are open-source/free/secure alternatives to Google Sheets which could be implemented/treated in the very same fashion.

**Comment by ssddanbrown** [comment]
Just a note on this, It's now possible for a user to switch the page editor, where permission exists, to somewhat achieve this.

**Comment by pdf** [comment]
I suspect I'm not alone in this, so I'd like to add my use-case.

My employer is a mix of technical and non-technical staff. I would dearly love to be able to enable markdown mode for everyone, so that pages are stored in a consistent format, and technical people can write docs efficiently as they're used to for all the other platforms they use. However it's just not feasible to enable markdown for the non-technical users without a WYSIWYG toolbar for them to click things on.

**Comment by ssddanbrown** [comment]
BookStack is now [managed on Codeberg](https://codeberg.org/bookstack/bookstack).

As part of this change, we're closing all existing GitHub issues and pull requests.
You'll instead find this issue on Codeberg here: 
https://codeberg.org/bookstack/bookstack/issues/315
  
Any further discussion for this will now be done on Codeberg.
Therefore this GitHub thread will be locked.
